### PR TITLE
use express-cluster for drop-in cluster support

### PR DESCRIPTION
**Pull request by jrsun**
This lightweight library (https://github.com/Flipboard/express-cluster) provides basic cluster support and some basic quality of life features like automatic respawning of worker processes, quitting the worker processes when the main process receives a SIGHUP, logging, etc.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -20,26 +21,32 @@ function composeImage(width, height, seed) {
...
-    res.end(buffer, 'binary');
-});
-
-app.listen(process.env.PORT || 3000, () => {
-    console.log('listening on http://localhost:3000');
-});
+cluster((worker) => {
+    const app = express();
+
+    app.get('/healthcheck', (req, res) => {
```
Comment (jrsun) [inline server.js]: should the healthcheck be handled by the master process?

File: server.js [hunk 2]
```diff
@@ -20,26 +21,32 @@ function composeImage(width, height, seed) {
...
+        res.end(buffer, 'binary');
+    });
+
+    activePort = process.env.PORT || 3000;
+    app.listen(activePort, () => {
+        console.log('worker ' + worker.id + ' is listening on port ' + activePort);
+    });
+}, {
+    'respawn': true, // workers will restart on failure
+    'verbose': true, // logs what happens to console
```
Comment (jrsun) [inline server.js]: it would be cool if we logged things somewhere

Comment (joshbuddy) [inline server.js]: Just print to stdout, it will go to the upstart logs

File: server.js (hunk 1 above)
Comment (joshbuddy) [inline server.js]: This is fine like this :)

File: server.js [hunk 3]
```diff
@@ -20,26 +21,32 @@ function composeImage(width, height, seed) {
...
-    res.set('Cache-Control', 'max-age=' + cacheTimeout);
-    res.set('Content-length', buffer.length);
-    res.type('png');
-    res.end(buffer, 'binary');
-});
-
-app.listen(process.env.PORT || 3000, () => {
-    console.log('listening on http://localhost:3000');
-});
+cluster((worker) => {
```
Comment (joshbuddy) [inline server.js]: Small thing, I'd put the cluster run command out to a separate file, `run.js`, and have it import the app. I'd also move all listening to there.

File: run.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const cluster = require('express-cluster');
+const app = require('./server.js');
+
+activePort = process.env.PORT || 3000;
```
Comment (joshbuddy) [inline run.js]: missing const here.

**Pull request merged**